modules/rostests/apitests/ws2_32/WSAAsync.c

```diff
@@ -227,12 +230,27 @@ START_TEST(WSAAsync)
         else
         {
             if (nSockNameRes != 0)
-                ok(0, "ERROR: getsockname function failed, expected %d error %d\n", 0, nSockNameRes);
+                ok(FALSE, "ERROR: getsockname function failed, expected 0 error %d\n", nSockNameRes);
             if (len != sizeof(addr_con_loc))
-                ok(0, "ERROR: getsockname function wrong size, expected %d returned %d\n", sizeof(addr_con_loc), len);
-            // FIXME: fails on WHS testbot
-            //if (addr_con_loc.sin_addr.s_addr != server_addr_in.sin_addr.s_addr)
-            //    ok(0, "ERROR: getsockname function wrong addr, expected %lx returned %lx\n", server_addr_in.sin_addr.s_addr, addr_con_loc.sin_addr.s_addr);
+                ok(FALSE, "ERROR: getsockname function wrong size, expected %Iu returned %d\n", sizeof(addr_con_loc), len);
+
+            if (addr_con_loc.sin_addr.s_addr == 0ul)
+            {
+                if (++Addr_con_locLoopCount >= MAX_LOOPCOUNT)
+                {
+                    ok(FALSE, "Giving up, on getsockname() (%u/%u), as addr_con_loc is not set yet\n",
+                       Addr_con_locLoopCount, MAX_LOOPCOUNT);
+                    goto done;
```

Comment:
Any reason why `break` didnt work here?

Reply:
No reason: I simply re-used `goto done`, as used earlier. As in 'error case', not simple 'end of loop'.
Just confirm if you do prefer `break` here.

Reply:
I'd prefer break from within the loop yes, seeing as the goto is used from before the loop, and break from within the loop.
It'd be fine with me to leave it as-is, if there is a strong preference for that.

Reply:
> break from within the loop

There is no `break` within _this_ loop.
Then, I leave it as is until further notice.